File: app/src/main/java/com/platdmit/simplecloudmanager/data/database/dao/BackupDao.java
package com.platdmit.simplecloudmanager.data.database.dao;

import com.platdmit.simplecloudmanager.data.database.entity.DbBackup;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Query;

@Dao
public interface BackupDao extends BaseDao<DbBackup>{
    @Query("SELECT * FROM dbbackup WHERE mServer = :serverId")
    List<DbBackup> getBackupsForServer(long serverId);
}

File: app/src/main/java/com/platdmit/simplecloudmanager/data/database/dao/BackupDao.kt
package com.platdmit.simplecloudmanager.data.database.dao

import androidx.room.Dao
import androidx.room.Query
import com.platdmit.simplecloudmanager.data.database.entity.DbBackup

@Dao
interface BackupDao : BaseDao<DbBackup> {
    @Query("SELECT * FROM dbbackup WHERE server = :serverId")
    fun getBackupsForServer(serverId: Long): List<DbBackup>?
}